### DISCUSSION TITLE
before test, async init database

### DISCUSSION THREAD

**Discussion by panla** [fastapi/fastapi#6645] [Questions] [answered]
- tortoise-orm
- fastapi
- pytest

i want to async init database before test.

- [X] create database before test
- [X] drop database after test
- [ ] async init data

the aim is 

```text
create database before test, and init data before test, and drop database after test.
```

**Comment by ArcLightSlavik** [comment]
https://tortoise-orm.readthedocs.io/en/latest/contrib/unittest.html#py-test

**Comment by panla** [comment]
@ArcLightSlavik 

no easy,

create and drop database, it is ok,

and the demand is after create database and create table, init some data.

such as 

```text
there is a model User, table users

after create database and create table,

insert some data into table users

after that, start test api
```

**Comment by ArcLightSlavik** [comment]
https://tortoise-orm.readthedocs.io/en/latest/examples/fastapi.html#example-fastapi

Take the code from there and put in your pytest fixture that initializes the db.

**Comment by YuriiMotov** [comment, marked as answer]
See also: https://fastapi.tiangolo.com/how-to/testing-database/
